src/Pos/Binary/Relay.hs

```diff
@@ -28,8 +28,13 @@ instance (Bi tag, Bi key) => Bi (ReqMsg key tag) where
     get = label "ReqMsg" $ liftM2 ReqMsg get get
 
 instance (Bi tag) => Bi (MempoolMsg tag) where
-    put MempoolMsg {..} = put mmTag
-    get = MempoolMsg <$> get
+    -- The extra byte is needed because time-warp doesn't work with
+    -- possibly-empty messages. 228 was chosen as homage to @pva701
+    put MempoolMsg {..} = put (228 :: Word8) >> put mmTag
+    get = label "MempoolMsg" $ do
+        x <- get @Word8
```

Comment:
And `getWord8`. But I started thinking that introducing `getWord8` instead of `get @Word8` may be bad because is not general API to use new function each time. Though in serialization it will be good because thus we can restrict type and save somebody from accidentally change `@Word8` to `@Word16`